runtime/oti/VMHelpers.hpp

```diff
@@ -1545,6 +1545,9 @@ class VM_VMHelpers
 			}
 			break;
 		case J9NtcBoolean:
+#if defined(OSX) && defined(J9AARCH64)
```

Comment:
In fact, I don't think I need to add the `#if defined(OSX) && defined(J9AARCH64)` guard for the return value part, because AArch64 OSX is a little endian platform.  The `#ifdef J9VM_ENV_LITTLE_ENDIAN` part gives the same result.
The change to the `J9NtcToFFI[]` table in BytecodeInterpreter.hpp is the essential part of this PR.

Which of the following would be better?
- to add a comment here as suggested, or
- to stop adding the `#if defined(OSX) && defined(J9AARCH64)` code for the return value part

Reply:
How does FFI work when you specify uint8? Does that mean that it's guaranteed to pass only 8 bits, or does it mean "do what C does with uint8"? If the latter, could we not simply use uint8 everywhere?

Reply:
@gacholio It is not clear to me which part of my change you are talking about by "specify uint8"?  The line `&ffi_type_uint8, /* J9NtcBoolean */` in BytecodeInterpreter?
The layout of arguments passed in the C stack will change on AArch64 macOS when you specify `&ffi_type_uint8` instead of `&ffi_type_uint32` there.  Libffi allocates a 1-byte slot in the C stack for `uint8`, and a 4-byte slot for `uint32`, and that will affect the stack offsets of other arguments.  That is what I need.
On AArch64 Linux, as an example of other platforms, the arguments on the C stack are always aligned at 8-byte boundary regardless of their sizes, and the stack offsets are not affected by `uint8` or `uint32` there.

Reply:
@0xdaryl I updated the commit.  I stopped adding the `#if defined(OSX) && defined(J9AARCH64)` code for the return value part.  I also added a comment in BytecodeInterpreter.hpp for the special handling on AArch64 macOS.

Reply:
I'm asking what using `&ffi_type_uint8` implies on other platforms - can we use this for boolean everywhere?

Reply:
@gacholio 
That is the topic of Issue #14574.  Using `&ffi_type_uint32` for `boolean` is a workaround introduced for a type inconsistency issue in the past.  The original BytecodeInterpreter code used to use `&ffi_type_uint8` for `boolean`.  I hope to revert the workaround on all the platforms, but I apply the change only to AArch64 macOS in this PR as a conservative start.

Reply:
Please read my summary in https://github.com/eclipse-openj9/openj9/issues/14574#issuecomment-1048442537 .